be/src/runtime/routine_load/routine_load_task_executor.cpp

```diff
@@ -320,8 +320,8 @@ void RoutineLoadTaskExecutor::exec_task(std::shared_ptr<StreamLoadContext> ctx,
 
         std::vector<RdKafka::TopicPartition*> topic_partitions;
         for (auto& kv : ctx->kafka_info->cmt_offset) {
-            RdKafka::TopicPartition* tp1 =
-                    RdKafka::TopicPartition::create(ctx->kafka_info->topic, kv.first, kv.second);
```

Comment:
Could you please explain why?

Reply:
The offsets you commit are the offsets of the messages you want to read next (not the offsets of the messages you did read last).